pkg/job/discovery.go

```diff
@@ -210,10 +212,19 @@ func getFilteredMetricDatas(
 	dimensionNameList []string,
 	m *config.Metric,
 ) []*model.CloudwatchData {
-	associator := newMetricsToResourceAssociator(dimensionRegexps, resources)
+	type resourceAssociator interface {
+		AssociateMetricToResource(cwMetric *cloudwatch.Metric) (*model.TaggedResource, bool)
+	}
```

Comment:
Would it make sense, in terms of readability, to move this interface type out of `getFilteredMetricDatas` and rather create `resourceAssociator` and pass that to `getFilteredMetricDatas` (instead of passing `ctx`, extracting flag, building associator etc.)?